src/software/control/src/main.rs

```diff
@@ -56,16 +61,25 @@ fn make_app_args() -> AppArgs {
                 .help("Serial port ID")
                 .takes_value(true),
         )
+        .arg(
+            Arg::with_name("file")
+                .short("f")
+                .long("file")
+                .help("Path to a recorded file")
+                .takes_value(true),
+        )
         .get_matches();
 
+    let source = match (matches.value_of("port"), matches.value_of("file")) {
+        (Some(p), _) => Source::Port(p.to_string()),
+        (None, Some(f)) => Source::File(f.to_string()),
+        (None, None) => panic!("You should provide either a serial port (-p) or a file (-f)"),
```

Comment:
I guess we will need to gracefully handle how the program exits at some point (printing an error and exit code > 0)

Reply:
Fixed